src/components/ui/GenesisStats.jsx

```diff
@@ -14,7 +14,7 @@ export function GenesisStats({ genesis = false }) {
 	const [status, setStatus] = useState(null);
 
 	useEffect(() => {
-		if (contracts) {
+		if (contracts && address) {
 			recountShare();
 		}
 	}, [contracts, timestamp]);
```

Comment:
should add address here as well, otherwise useEffect won't be run on address change and the condition is pointless